brian2/utils/caching.py

```diff
@@ -0,0 +1,92 @@
+'''
+Module to support caching of function results to memory (used to cache results
+of parsing, generation of state update code, etc.). Provides the `cached`
+decorator.
+'''
+
+import functools
+import collections
+
+
+class _CacheStatistics(object):
```

Comment:
I used this just for seeing that the caching actually worked. We could potentially use this in tests I guess?